### PR TITLE
gitea: init at 1.2.1

### PR DESCRIPTION
**Pull request by disassembler**
###### Motivation for this change
adds gitea package and service module

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bjornfor** [comment]
Could gitea share NixOS module with gogs? If we had "services.gogs.package = gogs|gitea"?

**Comment by disassembler** [comment]
It's a possibility at this point, but I'd be wary the further the fork diverges, the more things would be different. In this particular case, the systemd pre-start script needed to copy some locales over into the state directory the get the service to run. I also don't think upstream has any intention on keeping compatibility with gogs going forward.

File: nixos/modules/services/misc/gitea.nix [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        default = "";
+        description = "Configuration lines appended to the generated gitea configuration file.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.gitea = {
+      description = "gitea (Gitea Service)";
```
Comment (bjornfor) [inline nixos/modules/services/misc/gitea.nix]: Bikeshed: "Gitea self-hosted Git Service"?

File: nixos/modules/services/misc/gitea.nix [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        GITEA_WORK_DIR = cfg.stateDir;
+      };
+    };
+
+    users = mkIf (cfg.user == "gitea") {
+      extraUsers.gitea = {
+        description = "Gitea Service";
+        home = cfg.stateDir;
+        createHome = true;
+        shell = pkgs.bash;
```
Comment (bjornfor) [inline nixos/modules/services/misc/gitea.nix]: Does this user need a shell? (I don't think I've seen that option being used for daemon users before.)

**Review by bjornfor — APPROVED** [review]
Looks good.

File: nixos/modules/services/misc/gitea.nix (hunk 1 above)
Comment (jtojnar) [reply nixos/modules/services/misc/gitea.nix]: Gitea should be enough according to https://github.com/NixOS/nixpkgs/pull/28851/files#r136598912

File: nixos/modules/services/misc/gitea.nix (hunk 2 above)
Comment (disassembler) [reply nixos/modules/services/misc/gitea.nix]: good catch, copied the old module. This is gone and it still works.

File: nixos/modules/services/misc/gitea.nix (hunk 1 above)
Comment (disassembler) [reply nixos/modules/services/misc/gitea.nix]: done

**Comment by betaboon** [comment]
hello fellow nixtronauts,

I opened a gitea PR a while ago: https://github.com/NixOS/nixpkgs/pull/28851
It has some more fine grained config (eg forcing repositories to be private, disabling registration etc).
I think it still has value to add but i currently don't have the time to clean it up.

just wanted to let you know.

**Pull request merged**